keras_core/utils/dataset_utils.py

```diff
@@ -6,44 +11,481 @@
 def split_dataset(
     dataset, left_size=None, right_size=None, shuffle=False, seed=None
 ):
-    """Splits a dataset into a left half and a right half (e.g. train / test).
+    """Split a dataset into a left half and a right half (e.g. train / test).
 
     Args:
-        dataset: A `tf.data.Dataset` object, or a list/tuple of arrays with the
-            same length.
+        dataset: A `tf.data.Dataset or torchDataset`  object, or a list/tuple of arrays with the
+          same length.
         left_size: If float (in the range `[0, 1]`), it signifies
-            the fraction of the data to pack in the left dataset.
-            If integer, it signifies the number of samples to pack
-            in the left dataset. If `None`, it defaults to the complement
-            to `right_size`.
+          the fraction of the data to pack in the left dataset. If integer, it
```

Comment:
Use 4 space indent

Reply:
Done

Reply:
The indent is still 2 spaces.

You shouldn't actually need to modify the docstring at all, mind you. The original docstring was fine.